bitarray/_bitarray.c

```diff
@@ -87,13 +94,26 @@ static PyTypeObject Bitarraytype;
 
 #define BYTES(bits)  (((bits) == 0) ? 0 : (((bits) - 1) / 8 + 1))
 
-#define BITMASK(endian, i)  (((char) 1) << ((endian) ? (7 - (i)%8) : (i)%8))
+#define BITMASK(endian, i)  (((unsigned char) 1) << ((endian) ? (7 - (i)%8) : (i)%8))
 
 /* ------------ low level access to bits in bitarrayobject ------------- */
 
 #define GETBIT(self, i)  \
     ((self)->ob_item[(i) / 8] & BITMASK((self)->endian, i) ? 1 : 0)
 
+#if HAS_VECTORS
+/*
+ * Perform bitwise operation OP on 16 bytes of memory at a time.
+ */
+#define vector_op(A, B, OP) do {  \
+    vec __a, __b, __r;            \
+    memcpy(&__a, A, sizeof(vec)); \
```

Comment:
@diamondman It's an unaligned load, not a copy. You have to load data into registers anyway. The loop in @udoprog's last post is about as good as it gets without unrolling.

I think this is more than good enough for a start. On non x86 there may be a room for improvement, but I wouldn't complicate the code without testing it on an ARM or PPC machine first. And in any case, I think this is enough for one PR.